sdk/core/core/inc/az_json_data.h

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef _az_JSON_DATA_H
+#define _az_JSON_DATA_H
+
+#include <az_json_token.h>
+#include <az_mut_span.h>
+#include <az_span.h>
+
+#include <stdbool.h>
+
+#include <_az_cfg_prefix.h>
+
+typedef struct az_json_data az_json_data;
+
+typedef struct {
+  struct {
+    az_json_data const * begin;
+    size_t size;
+  } _detail;
+} az_json_array;
+
+AZ_NODISCARD AZ_INLINE size_t az_json_array_size(az_json_array const a) { return a._detail.size; }
+
+typedef struct az_json_object_member az_json_object_member;
```

Comment:
Can we avoid this by rearranging declarations?

Reply:
No, there are circular dependencies.